### PR TITLE
Make xpass failure again

### PR DESCRIPTION
**Pull request by TanyaAgarwal28**
**Problem:**

Some years ago, a change was made that caused `xpass` not to be considered a failure. However, there have been multiple cases where it was missed that a test was fixed, and this is not an acceptable default behavior.

**Solution:**

This PR aims to address the issue by modifying the behavior of `xpass` in the following way: In this strict parameter of xfail is set to true by default and it will ensure xpass is either a warning or a failure in any case.

**Checklist:**

- [X] Include documentation for the proposed change.
- [ ] Include new tests or update existing tests, where applicable.
- [X] Allow maintainers to push and squash when merging my commits.

This change closes issue #11467.

Changelog: Created a new changelog file named `11467.feature.rst` and added the entry for this change.

Author: Tanya Agarwal

### CODE REVIEW

File: src/_pytest/skipping.py [hunk 1]
```diff
@@ -208,7 +207,7 @@ def evaluate_xfail_marks(item: Item) -> Optional[Xfail]:
     """Evaluate xfail marks on item, returning Xfail if triggered."""
     for mark in item.iter_markers(name="xfail"):
         run = mark.kwargs.get("run", True)
-        strict = mark.kwargs.get("strict", item.config.getini("xfail_strict"))
+        strict = mark.kwargs.get(item.config.getini("xfail_strict"), True)
```
Comment (RonnyPfannschmidt) [inline src/_pytest/skipping.py:210]: in order to ensure backward compatibility we cannot go straight from  non-strict to strict
instead we have to start by warning if strict was not set to true or false

the warning should indicate that a future major release of pytest will change the default from False to True
and recommend to use struct=True as default and a plugin for actually flaky tests

Comment (TanyaAgarwal28) [reply src/_pytest/skipping.py:210]: Sure

**Comment by TanyaAgarwal28** [comment]
Sorry I'm closing this PR due to misunderstaning of a solution. I have created an another PR.
Make xpass failure again warning #11467

**Comment by TanyaAgarwal28** [comment]
Closes

**Comment by TanyaAgarwal28** [comment]
Please find below the latest open PR for this same issue with the changes suggested by you as well.
#11499 
@RonnyPfannschmidt

**Pull request closed without merging**